rust_src/src/process.rs

```diff
@@ -84,15 +89,36 @@ impl LispProcessRef {
     }
 }
 
+/// This is how commands for the user decode process arguments.  It
+/// accepts a process, a process name, a buffer, a buffer name, or nil.
+/// Buffers denote the first process in the buffer, and nil denotes the
+/// current buffer.
+pub fn get_process(process: LispObject) -> LispProcessRef {
+    let process = if process.is_string() {
+        let p = lisp_get_process(process);
+        if p.is_nil() {
+            get_buffer_process(process)
+        } else {
+            p
+        }
+    } else if process.is_nil() {
```

Comment:
That should not be an else. It should be a stand alone if.

Reply:
Good spot.